pkg/flare/flare.go

```diff
@@ -86,6 +90,15 @@ func SendFlareWithHostname(archivePath string, caseID string, email string, host
 	client := mkHTTPClient()
 	r, err := client.Do(request)
 
+	// Handle redirects manually. Go's http.Client doesn't know how to do it when it can't seek
+	// back to the beginning of the body. Since we are using a pipe, seeking isn't possible.
+	// Re-sending a POST is only legal for status 307, so we only need to check for that code.
+	if r != nil && r.StatusCode == 307 && redirectHops < 5 {
+		url = r.Header.Get("Location")
+		redirectHops++
+		goto redirect
```

Comment:
It's not that I'm against `goto`, but it looks like this could also be done with other methods, which could be easier to follow (memory/routines speaking), maybe with a channel receiving URLs to connect to?

Reply:
It can be done with a function call in a loop, which seems to most readable alternative to me. I didn't do it this way to keep the patch small, but I can change it. 👍

Reply:
I thought about the for loop but I was afraid I've missed something obvious 😌. I think a biggest patch is definitely worth it for clarity in the long term 👍